crates/sqlexec/src/dispatch/external.rs

```diff
@@ -592,27 +592,47 @@ impl<'a> ExternalDispatcher<'a> {
             .transpose()?
             .unwrap_or(FileCompressionType::UNCOMPRESSED);
 
-        let ft: FileType = file_type.parse()?;
-        let ft: Arc<dyn FileFormat> = match ft {
-            FileType::CSV => Arc::new(
-                CsvFormat::default()
-                    .with_file_compression_type(compression)
-                    .with_schema_infer_max_rec(Some(20480)),
-            ),
-            FileType::PARQUET => Arc::new(ParquetFormat::default()),
-            FileType::JSON => {
-                Arc::new(JsonFormat::default().with_file_compression_type(compression))
-            }
-            _ => return Err(DispatchError::InvalidDispatch("Unsupported file type")),
-        };
-
-        let accessor = ObjStoreAccessor::new(access)?;
-        let objects = accessor.list_globbed(path).await?;
-
-        let state = self.df_ctx.state();
-        let provider = accessor.into_table_provider(&state, ft, objects).await?;
+        let accessor = ObjStoreAccessor::new(access.clone())?;
 
-        Ok(provider)
+        match file_type {
+            "csv" => Ok(accessor
```

Comment:
Apparently the file type is getting stored with with quotes, so matching on `"\"csv\""` works...

Reply:
I believe it's not getting stored as a quote but just printed into a quoted form when it's read which is easy enough clean up.